### DISCUSSION TITLE
下载导出问题

### DISCUSSION THREAD

**Discussion by JackSaber** [hhyo/Archery#2117] [Q&A]
当SQL查询数据超大时，比如20w行以上的数据，在export时，会崩溃或卡死，有没有好的解决方案？或者优化呢

**Comment by LeoQuote** [comment]
这方面可以通过异步的方式解决, 但是这个功能并非 archery 的核心功能, 所以可能不会去做, 但是欢迎pr, 谢谢.

**Comment by cxheny** [comment]
我也苦于没有该项功能，目前自己改了一些代码，还在测试中。

**Reply by i-xingchen** [reply]
大佬，有没有好的办法，求指导

**Reply by cxheny** [reply]
<img width="1095" alt="微信图片_20240223110027" src="https://github.com/hhyo/Archery/assets/40221697/9ea7c44f-2da6-4c00-8a3d-0e8e32f1d02b">
内部已经上线使用了，没时间做PR

**Comment by Wondermique** [comment]
针对1.9.1二开版本，还在最后做一些细节优化，在查询页面通过按钮切换，按钮显示与否取决于系统配置内是否开启了离线导出，提交页面会检查执行计划及是否为查询语句，目前测试过80W的数据导出，可正常导出下载。
![image](https://github.com/hhyo/Archery/assets/133197875/9fff456f-22dc-49ae-8afb-f957792c811b)
![image](https://github.com/hhyo/Archery/assets/133197875/467f3f35-c932-44ab-ab09-f5582621d304)
![image](https://github.com/hhyo/Archery/assets/133197875/107ff1ca-b6b6-4650-8ad7-92ed37363614)
![image](https://github.com/hhyo/Archery/assets/133197875/5645c822-57a0-4f4f-a8eb-2e32ba25f69e)
![image](https://github.com/hhyo/Archery/assets/133197875/4193667c-1a79-41bc-ae26-5ad3d6c75833)

**Reply by i-xingchen** [reply]
太牛了，我的老哥，这个功能对我们来说很重要，想请问下功能啥时候可以支持，感激不尽，或者指导下代码怎么改造

**Reply by Wondermique** [reply]
> 太牛了，我的老哥，这个功能对我们来说很重要，想请问下功能啥时候可以支持，感激不尽，或者指导下代码怎么改造

年后会尽量找时间PR，目前还无法确定具体时间。

**Reply by jirba96** [reply]
太牛逼了，我目前需要这个功能，请问什么时候能支持一下
